awswrangler/data_api/rds.py

```diff
@@ -42,42 +57,75 @@ def __init__(
         retries: int = 30,
         boto3_session: Optional[boto3.Session] = None,
     ) -> None:
+        super().__init__()
+
+        self.client = _utils.client(service_name="rds-data", session=boto3_session)
+
         self.resource_arn = resource_arn
         self.database = database
         self.secret_arn = secret_arn
         self.wait_config = _connector.WaitConfig(sleep, backoff, retries)
-        self.client = _utils.client(service_name="rds-data", session=boto3_session)
-        self.results: Dict[str, Dict[str, Any]] = {}
-        logger: logging.Logger = logging.getLogger(__name__)
-        super().__init__(self.client, logger)
+        self.results: Dict[str, Union["ExecuteStatementResponseTypeDef", "BatchExecuteStatementResponseTypeDef"]] = {}
 
-    def _execute_statement(self, sql: str, database: Optional[str] = None) -> str:
-        if database is None:
-            database = self.database
+    def close(self) -> None:
+        """Close underlying endpoint connections."""
+        self.client.close()
 
+    def begin_transaction(self, database: Optional[str] = None, schema: Optional[str] = None) -> str:
+        """Start an SQL transaction."""
+        kwargs = {}
+        if database:
+            kwargs["database"] = database
```

Comment:
If database is None, should we not use the value supplied when instantiating the class (`self.database`)?